core/src/main/java/com/novoda/noplayer/internal/exoplayer/AvailableAdverts.java

```diff
@@ -0,0 +1,41 @@
+package com.novoda.noplayer.internal.exoplayer;
+
+import com.google.android.exoplayer2.source.ads.AdPlaybackState;
+import com.novoda.noplayer.AdvertBreak;
+
+import java.util.List;
+
+final class AvailableAdverts {
+
+    private AvailableAdverts() {
+        // Utility class.
+    }
+
+    /**
+     * Transforms all skipped adverts to available adverts.
+     * <p>
+     * WARNING: This relies on mutation to achieve what would not be possible
+     * as the exo-player {@link AdPlaybackState} does not allow transitioning from
+     * Skipped to Available adverts.
+     *
+     * @param currentPositionInMillis The position after which all adverts will transition from Skipped to Available.
+     * @param advertBreaks            The client representation of the adverts, our source of truth.
+     * @param adPlaybackState         The {@link AdPlaybackState} to mutate with the new states.
+     * @return The {@link AdPlaybackState} with the new available states.
+     */
+    static AdPlaybackState from(long currentPositionInMillis, List<AdvertBreak> advertBreaks, AdPlaybackState adPlaybackState) {
```

Comment:
I think we should not return anything and have a clearer method name to highlight what happens.

Reply:
a1a8c63